File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task145_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <vector>
#include <algorithm>
#include <climits>

using namespace std;

int maxSubarraySum(vector<int>& arr) {
    int maxSum = INT_MIN;
    int currentSum = 0;
    for (int num : arr) {
        currentSum = max(num, currentSum + num);
        maxSum = max(maxSum, currentSum);
    }
    return maxSum;
}

int main() {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    cout << maxSubarraySum(arr) << endl;
    return 0;
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task145_CLAUDE_claude-3-5-sonnet-20240620.java

import java.util.Scanner;

public class MaxSubarraySum {
    public static int maxSubarraySum(int[] arr) {
        int maxSum = Integer.MIN_VALUE;
        int currentSum = 0;
        for (int num : arr) {
            currentSum = Math.max(num, currentSum + num);
            maxSum = Math.max(maxSum, currentSum);
        }
        return maxSum;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }
        System.out.println(maxSubarraySum(arr));
        scanner.close();
    }
}
